packages/geolocator/tizen/src/location_manager.cc

```diff
@@ -0,0 +1,178 @@
+                          &manager_for_current_location_);
+}
+
+LocationManager::~LocationManager() {
+  location_manager_destroy(manager_);
+  location_manager_destroy(manager_for_current_location_);
+}
+
+bool LocationManager::IsLocationServiceEnabled() {
+  bool gps_enabled = false;
+  int ret_gps =
+      location_manager_is_enabled_method(LOCATIONS_METHOD_GPS, &gps_enabled);
+  if (LOCATIONS_ERROR_NONE != ret_gps) {
+    throw LocationManagerError(ret_gps);
+  }
+
+  bool wps_enabled = false;
+  int ret_wps =
+      location_manager_is_enabled_method(LOCATIONS_METHOD_WPS, &wps_enabled);
+  if (LOCATIONS_ERROR_NONE != ret_wps) {
+    throw LocationManagerError(ret_wps);
+  }
+  return gps_enabled || wps_enabled;
+}
+
+Position LocationManager::GetLastKnownPosition() {
+  double altitude = 0.0;
+  double latitude = 0.0;
+  double longitude = 0.0;
+  double climb = 0.0;
+  double direction = 0.0;
+  double speed = 0.0;
+  double horizontal = 0.0;
+  double vertical = 0.0;
+  location_accuracy_level_e level = LOCATIONS_ACCURACY_NONE;
+  time_t timestamp = 0;
+
+  int ret = location_manager_get_last_location(
+      manager_, &altitude, &latitude, &longitude, &climb, &direction, &speed,
+      &level, &horizontal, &vertical, &timestamp);
+  if (LOCATIONS_ERROR_NONE != ret) {
+    throw LocationManagerError(ret);
+  }
+  Position position;
+  position.longitude = longitude;
+  position.latitude = latitude;
+  position.timestamp = timestamp;
+  position.accuracy = horizontal;
+  position.altitude = altitude;
+  position.heading = direction;
+  position.speed = speed;
+
+  return position;
+}
+
+void LocationManager::GetCurrentPosition(
+    LocationCallback location_callback,
+    LocationErrorListener location_error_listener) {
+  location_callback_ = location_callback;
+  location_error_listener_ = location_error_listener;
+  int ret = location_manager_request_single_location(
+      manager_for_current_location_, 5,
+      [](location_error_e error, double latitude, double longitude,
+         double altitude, time_t timestamp, double speed, double direction,
+         double climb, void *user_data) {
+        LocationManager *self = static_cast<LocationManager *>(user_data);
+
+        if (error != LOCATIONS_ERROR_NONE && self->location_error_listener_) {
+          self->location_error_listener_(LocationManagerError(error));
+        } else if (self->location_callback_) {
+          Position position;
+          position.latitude = latitude;
+          position.longitude = longitude;
+          position.altitude = altitude;
+          position.timestamp = timestamp;
+          position.speed = speed;
+          position.heading = direction;
+
+          self->location_callback_(position);
+        }
+
+        self->location_callback_ = nullptr;
+        self->location_error_listener_ = nullptr;
+      },
+      this);
+  if (LOCATIONS_ERROR_NONE != ret) {
+    throw LocationManagerError(ret);
```

Comment:
When do you use an exception and when do you use the `LastError` pattern? Please choose only one if possible.

Reply:
I removed `LastError` pattern